controllers/precacheFsm.go

```diff
@@ -59,6 +59,20 @@ const (
 func (r *ClusterGroupUpgradeReconciler) precachingFsm(ctx context.Context,
 	clusterGroupUpgrade *ranv1alpha1.ClusterGroupUpgrade) error {
 
+	specCondition := meta.FindStatusCondition(clusterGroupUpgrade.Status.Conditions, "PrecacheSpecValid")
+	if specCondition == nil || specCondition.Status == metav1.ConditionFalse {
+		allManagedPoliciesExist, _, managedPoliciesPresent, err := r.doManagedPoliciesExist(ctx, clusterGroupUpgrade)
+		if err != nil {
+			return err
+		}
+
+		if allManagedPoliciesExist {
+			err = r.reconcileResources(ctx, clusterGroupUpgrade, managedPoliciesPresent)
```

Comment:
Why do we also need this part?

Reply:
Yeah...I'm not sure I understand this PR, I must be missing context.
If precaching is for 'let's stage the artifacts for the upcoming upgrade', what is this reconcile policies call fixing?

Reply:
Precaching needs information from the managed policies, but it also needs for the policies to be in the same namespace as the CGU. reconcileResources is creating copies of the managed policies in the namespace of the CGU, so pre-caching can access them. I discussed with Vitaly about the possibility of using doManagedPoliciesExist to get the namespaces of the managed policies instead of calling the reconcileResources.

Reply:
I think it's much better now.
Thanks folks